diff --git a/grib_api/src/grib_errors.c b/grib_api/src/grib_errors.c
index 76edec7..ce0bc19 100644
--- a/grib_api/src/grib_errors.c
+++ b/grib_api/src/grib_errors.c
@@ -99,44 +99,12 @@ void grib_check(const char* call,const char*  file,int line,int e,const char* ms
 		} else {
 			grib_context_log(c,GRIB_LOG_ERROR,"%s",grib_get_error_message(e));
 		}
-        grib_exit(e);
+        exit(e);
     }
 }
 
-grib_fail_proc _grib_fail = NULL;
-
-void grib_set_fail_proc(grib_fail_proc p) 
-{
-	_grib_fail = p;
+void grib_fail(const char* expr,const char* file,int line) {
+   fprintf(stderr,"%s at line %d: assertion failure Assert(%s)\n",file,line,expr);
+   abort();
 }
 
-void grib_fail_default(const char* expr, const char* file, int line) 
-{
-	fprintf(stderr, "%s at line %d: assertion failure Assert(%s)\n", file, line, expr);
-	abort();
-}
-
-void grib_fail(const char* expr,const char* file,int line)
-{
-	if (_grib_fail) {
-		_grib_fail(expr, file, line);
-	} else {
-		grib_fail_default(expr, file, line);
-	}
-}
-
-grib_exit_proc _grib_exit = NULL;
-
-void grib_set_exit_proc(grib_exit_proc p)
-{
-	_grib_exit = p;
-}
-
-void grib_exit(int code)
-{
-	if (_grib_exit) {
-		_grib_exit(code);
-	} else {
-		exit(code);
-	}
-}
\ No newline at end of file
